app/src/main/java/com/my/journeyplanner/presenters/MainPresenter.kt

```diff
@@ -0,0 +1,53 @@
+package com.my.journeyplanner.presenters
+
+import android.util.Log
+import com.my.journeyplanner.apiService
+import com.my.journeyplanner.models.JourneyPlannerDisambiguationResult
+import com.my.journeyplanner.views.main.MainContract
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+class MainPresenter(private val view: MainContract.View) : MainContract.Presenter {
+    override fun onChangTimeClicked() {
+
+    }
+
+    override fun onEditPreferencesClicked() {
+
+    }
+
+    override fun onPlanMyJourneyClicked(): Call<JourneyPlannerDisambiguationResult> {
+        val call = apiService.getJourneyResults(
+            view.getFromLocation().text.toString(),
+            view.getToLocation().text.toString()
+        )
+
+        call.enqueue(object : Callback<JourneyPlannerDisambiguationResult> {
+            override fun onFailure(call: Call<JourneyPlannerDisambiguationResult>, throwable: Throwable) {
+                Log.e(TAG, "Other error: ${throwable.message}")
+                Log.e(TAG, "Other error: ${throwable.stackTrace.forEach { println(it) }}")
```

Comment:
mark TODOs with the intended future changes

Reply:
Added:
/*            TODO("Default adapter treats response code 300 Multiple Options as unsuccessful and")
            TODO("puts response in the errorBody but is actually a valid response.")
            TODO("Implement custom adapter to parse the valid response in the errorBody")   */